src/server/handlers/dynamic.js

```diff
@@ -24,7 +24,22 @@ export default ({ server, config }) => {
       // Set a cache key
       const currentPath = request.url.pathname || '/'
       const isPreview = Boolean(request.query && request.query.tapestry_hash)
-      const cacheKey = normaliseUrlPath(currentPath)
+      let cacheKey = normaliseUrlPath(currentPath)
+
+      // Run cacheKeyHandler function if provided by client
+      const cacheKeyHandler = config.cacheKeyHandler
+      let newCacheKey = cacheKey
+      if (typeof cacheKeyHandler === 'function') {
+        try {
+          newCacheKey = cacheKeyHandler({ ...request }, cacheKey)
```

Comment:
Could we just define `newCacheKey` here? e.g.

```js
if (typeof cacheKeyHandler === 'function') {
  try {
    let newCacheKey = cacheKeyHandler({ ...request }, cacheKey)
    if (typeof newCacheKey !== 'string') {
      throw `cacheKeyHandler() return value: expected "string" but received "${typeof newCacheKey}"`
    }
    cacheKey = newCacheKey
  } catch (e) {
    log.error(`cacheKeyHandler() error: ${e}`)
  }
}
```

Also, might we need to `return throw`? Or will the throw cancel the following execution of the `cacheKey = newCacheKey`?

Reply:
It seems throw immediately leaves the try block, [source](https://stackoverflow.com/questions/26067190/do-i-need-to-return-after-throw-in-javascript)